ugsync/src/main/java/org/apache/ranger/ldapusersync/process/LdapUserGroupBuilder.java

```diff
@@ -52,1240 +50,1161 @@
+                                continue;
+                            }
+
+                            String userFullName = userEntry.getNameInNamespace();
+                            String userName     = (String) userNameAttr.get();
+
+                            if (userName == null || userName.trim().isEmpty()) {
+                                LOG.info("{} empty for entry {}, skipping sync", userNameAttribute, userEntry.getNameInNamespace());
+                                continue;
+                            }
+
+                            Attribute timeStampAttr = attributes.get("uSNChanged");
+                            if (timeStampAttr != null) {
+                                String uSNChangedVal        = (String) timeStampAttr.get();
+                                long   currentDeltaSyncTime = Long.parseLong(uSNChangedVal);
+                                LOG.info("uSNChangedVal = {} and currentDeltaSyncTime = {}", uSNChangedVal, currentDeltaSyncTime);
+                                if (currentDeltaSyncTime > highestdeltaSyncUserTime) {
+                                    highestdeltaSyncUserTime = currentDeltaSyncTime;
+                                }
+                            } else {
+                                timeStampAttr = attributes.get("modifytimestamp");
+                                if (timeStampAttr != null) {
+                                    String timeStampVal         = (String) timeStampAttr.get();
+                                    Date   parseDate            = dateFormat.parse(timeStampVal);
+                                    long   currentDeltaSyncTime = parseDate.getTime();
+                                    LOG.info("timeStampVal = {} and currentDeltaSyncTime = {}", timeStampVal, currentDeltaSyncTime);
+                                    if (currentDeltaSyncTime > highestdeltaSyncUserTime) {
+                                        highestdeltaSyncUserTime = currentDeltaSyncTime;
+                                        deltaSyncUserTimeStamp   = timeStampVal;
+                                    }
+                                }
+                            }
+
+                            // Get all the groups from the group name attribute of the user only when group search is not enabled.
+                            if (!groupSearchEnabled) {
+                                for (String useGroupNameAttribute : userGroupNameAttributeSet) {
+                                    Attribute userGroupfAttribute = userEntry.getAttributes().get(useGroupNameAttribute);
+                                    if (userGroupfAttribute != null) {
+                                        NamingEnumeration<?> groupEnum = userGroupfAttribute.getAll();
+                                        while (groupEnum.hasMore()) {
+                                            String groupDN = (String) groupEnum.next();
+                                            if (LOG.isDebugEnabled()) {
```

Comment:
Consider removing `if (LOG.isDebugEnabled()) {`, unless arguments to log calls are expensive to compute.